### PR TITLE
Added the CFour Job Adapter

### PR DESCRIPTION
**Pull request by alongd**
Adding support for [CFour](http://slater.chemie.uni-mainz.de/cfour/index.php) (Coupled-Cluster techniques for Computational Chemistry)

### CODE REVIEW

File: arc/mainTest.py [hunk 1]
```diff
@@ -95,7 +95,8 @@ def test_as_dict(self):
                                              'method_type': 'dft',
                                              'software': 'gaussian'},
                          'e_confs': 5.0,
-                         'ess_settings': {'gaussian': ['local', 'server2'],
+                         'ess_settings': {'cfour': ['local'],
```
Comment (kfir4444) [inline arc/mainTest.py:98]: consider adding more tests here, we will soon need to check psi4 as well on `local`, maybe add it to the issues?

**Review by kfir4444 — DISMISSED** [review]

**Review by kfir4444 — COMMENTED** [review]
Good job @alongd!
Please rebase and push again, to avoid the conflicts.
otherwise approved.

**Comment by alongd** [comment]
@kfir4444, the branch was rebased, please take another look.

Also, please see if any of the commits here introduce modifications that we need to replicate in the Psi4 PR.

**Comment by kfir4444** [comment]
Thanks again @alongd!
Please note that the CI did not pass. Seems like a quick fix though.

**Comment by kfir4444** [comment]
@alongd may I suggest a fix for the failing test?
if you write in ` /home/runner/work/ARC/ARC/arc/levelTest.py, line 206, in test_determine_compatible_ess`:
`self.assertEqual(sorted(level_1.compatible_ess), sorted(['gaussian', 'molpro', 'cfour']))`
instead of:
`self.assertEqual(level_1.compatible_ess, ['gaussian', 'molpro', 'cfour'])`
than you will get correct results regardless of the order.

**Review by kfir4444 — APPROVED** [review]
Thanks @alongd!

**Comment by kfir4444** [comment]
Great! @alongd

**Review by kfir4444 — APPROVED** [review]

**Pull request merged**